dart/lib/src/sentry_client.dart

```diff
@@ -92,11 +101,27 @@ class SentryClient {
       platform: event.platform ?? sdkPlatform,
     );
 
-    if (event.throwable != null && event.exception == null) {
+    if (stackTrace != null &&
+        event.throwable == null &&
+        event.exception == null) {
+      final frames = _stackTraceFactory.getStackFrames(stackTrace);
+
+      if (frames != null && frames.isNotEmpty) {
+        event = event.copyWith(stackTrace: SentryStackTrace(frames: frames));
+      }
+    } else if (event.throwable != null && event.exception == null) {
       final sentryException = _exceptionFactory
-          .getSentryException(event.throwable, stackTrace: event.stackTrace);
+          .getSentryException(event.throwable, stackTrace: stackTrace);
 
       event = event.copyWith(exception: sentryException);
+    } else if (stackTrace == null &&
+        _options.attachStackTrace &&
+        event.exception == null) {
+      final frames = _stackTraceFactory.getStackFrames(StackTrace.current);
+
+      if (frames != null && frames.isNotEmpty) {
+        event = event.copyWith(stackTrace: SentryStackTrace(frames: frames));
+      }
```

Comment:
I suggest extracting this to a method on `_stackTraceFactory`, here and `getSentryException` method kinda does the same thing to convern a dynamic stackTrace to a SentryStackTrace and looks like both are different.

what we need is:

if theres an `exception`, do not set `event.stackTrace`, end.

if there's a `throwable`, do not set `event.stackTrace`, because the exception already has most probably an exception, end.

if theres not a `throwable` nor a `exception`, lets set the `event.stackTrace` only if `attachStacktrace` is enabled.

for all the cases above, if the `stackTrace` is not null, use it, if its null, use `stackTrace.current`

Reply:
yeah it was messy. I updated to 
```dart
    if (event.exception != null) return event;

    if (event.throwable != null) {
      final sentryException = _exceptionFactory
          .getSentryException(event.throwable, stackTrace: stackTrace);

      return event.copyWith(exception: sentryException);
    }

    if (stackTrace != null || _options.attachStackTrace) {
      stackTrace ??= StackTrace.current;
      final frames = _stackTraceFactory.getStackFrames(stackTrace);

      if (frames != null && frames.isNotEmpty) {
        event = event.copyWith(stackTrace: SentryStackTrace(frames: frames));
      }
    }

    return event;
```
I'm not sure we really need to create another method. WDYT ?

Reply:
```
if (stackTrace != null || _options.attachStackTrace) {
      stackTrace ??= StackTrace.current;
```

still wrong.
the code looks better now, I'd still extract to a method like I said before, DRY